src/lib/Interpreter.hs

```diff
@@ -209,10 +209,6 @@ evalOp expr = mapM evalAtom expr >>= \case
     let failedCast = error $ "Cast not implemented: " ++ pprint sourceTy ++
                              " -> " ++ pprint destTy
     case (sourceTy, destTy) of
-      (IdxRepTy, TC (Fin n)) -> return $ Con $ FinVal n x
-      (TC (Fin _), IdxRepTy) -> do
-        let Con (FinVal _ ord) = x
-        return ord
```

Comment:
Why are these no longer needed?

Reply:
You no longer need a `CastOp` to construct a `Fin` value: it is now `Con (Newtype (TC (Fin n)) ordinal_atom`. And to extract the payload you use the projection, as explained in the comment above.